hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -424,6 +406,9 @@ private void internalInit(
 				/* This was an upgrade, discard now-obsolete preparation state */
 				networkCtx().discardPreparedUpgradeMeta();
 				dualState.setFreezeTime(null);
+				System.out.println("START OVER START OVER START OVER");
+				networkCtx().setFirstConsTimeOfCurrentBlock(null);
+				networkCtx().setCurrentBlockHash(new Hash());
```

Comment:
Is the current block hash **ever** available? (In Ethereum, `blockhash(block.number)` is always `0x`.)

If it's never available, I don't know if we need to track it explicitly.

Reply:
Current hash is impossible to figure out in the EVM. The execution of the program winds up being one of the inputs. It's always getting the hash from 1 to 255 blocks behind